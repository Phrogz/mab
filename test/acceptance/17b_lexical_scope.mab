p( "Test #17b" )
setSlot( "x", 'global' )
setSlot( "foo", method(
	setSlot( "x", 'foo' )
	p( x )
	bar
) )
setSlot( "bar", method(
	p( x )
) )
foo

setSlot( 'recurse', method( depth,
	if( depth < 4,
		if( depth == 1, setSlot( "a", 'a' ) )
		if( depth == 2, setSlot( "b", 'b' ) )
		if( depth == 3, setSlot( "c", 'c' ) )
		p( "#{a}#{b}#{c}#{d}" interpolate )
		recurse( depth + 1 )
	)
) )
setSlot( 'a', '-' )
setSlot( 'b', '-' )
setSlot( 'c', '-' )
setSlot( 'd', '-' )
recurse( 0 )