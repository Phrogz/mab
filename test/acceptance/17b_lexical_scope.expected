Test #17b
foo
global
----
a---
-b--
--c-
